.circleci/config.yml

```diff
@@ -491,9 +491,9 @@ commands:
                 name: Wait for memcached
                 command: dockerize -wait tcp://localhost:11211 -timeout 1m
       - run:
-          name: Set environment variables
+          name: Set environment variables and install olympia
           command: |
-            echo export PYTHONPATH=src >> $BASH_ENV
+            pip install -e .
```

Comment:
This... seems like an unexpected change.  Previously we were exporting `PYTHONPATH` and now we're doing `pip install` one every job`?  If it's necessary for the localization job surely we don't need it for all the other jobs too?

Reply:
I am not sure we should be doing this for every job, but the `PYTHONPATH=src` was a hack that we should aim to get rid of for consistency.

Reply:
I think ideally we should do something like:
```diff
diff --git a/Makefile-docker b/Makefile-docker
index 53ed608c40..7495ea57bb 100644
--- a/Makefile-docker
+++ b/Makefile-docker
@@ -80,12 +80,17 @@ cleanup_python_build_dir:
 	# Work arounds "Multiple .dist-info directories" issue.
 	rm -rf /deps/build/*
 
+.PHONY: install_python_olympia_module
+install_python_olympia_module:
+	# pep 517 mode (the default) breaks editable install in our project. https://github.com/mozilla/addons-server/issues/16144
+	$(PIP_COMMAND) install --no-use-pep517 -e .
+
 .PHONY: install_python_codestyle_dependencies
 install_python_codestyle_dependencies:
 	$(PIP_COMMAND) install --progress-bar=off --no-deps --exists-action=w -r requirements/codestyle.txt
 
 .PHONY: install_python_test_dependencies
-install_python_test_dependencies:
+install_python_test_dependencies: install_python_olympia_module
 	$(PIP_COMMAND) install --progress-bar=off --no-deps --exists-action=w -r requirements/system.txt
 	$(PIP_COMMAND) install --progress-bar=off --no-deps --exists-action=w -r requirements/prod.txt
 	$(PIP_COMMAND) install --progress-bar=off --no-deps --exists-action=w -r requirements/tests.txt
@@ -94,8 +99,6 @@ install_python_test_dependencies:
 install_python_dev_dependencies: install_python_test_dependencies install_python_codestyle_dependencies
 	$(PIP_COMMAND) install --progress-bar=off --no-deps --exists-action=w -r requirements/dev.txt
 	$(PIP_COMMAND) install --progress-bar=off --no-deps --exists-action=w -r requirements/docs.txt
-	# pep 517 mode (the default) breaks editable install in our project. https://github.com/mozilla/addons-server/issues/16144
-	$(PIP_COMMAND) install --no-use-pep517 -e .
 
 .PHONY: install_node_dependencies
 install_node_dependencies: install_node_js copy_node_js
```

+ remove `echo export PYTHONPATH=src >> $BASH_ENV` from `.circleci/config.yml` and don't add the `pip install -e .`.

Reply:
Just curious, if the result of running the `pip install -e .` is _essentially_ to set the PYTHONPATH appropriately, and we were already doing that export in the setup_container part of our pipeline, then what exactly is the downside of running `pip install` here?

I'm not opposed to doing it somewhere else, it just seems like the alternative is harder to find and thus harder to understand.

Reply:
`pip install -e` installs a special file in the regular `PYTHONPATH` that helps `python` understand where the `olympia` module is.

My proposal removes the duplicate `pip install -e .` call that your patch introduces, and leaves only a single place where we do this, so I think it's valuable on top of removing the `PYTHONPATH=src` (which had a lot more consequences, since it forced _all_ modules to be looked at in that directory in addition to the regular directories where modules are found)

Reply:
Ah I didn't even see the sneaky duplicated call there. Okay, this makes more sense. I'll apply this change to the PR.